gitlab/resource_gitlab_project.go

```diff
@@ -206,11 +240,16 @@ func resourceGitlabProjectUpdate(d *schema.ResourceData, meta interface{}) error
 		options.SnippetsEnabled = gitlab.Bool(d.Get("snippets_enabled").(bool))
 	}
 
-	log.Printf("[DEBUG] update gitlab project %s", d.Id())
+	if *options != (gitlab.EditProjectOptions{}) {
```

Comment:
what is this? Why do we need that if?

Reply:
I guess this is meant to only invoke `client.Projects.EditProject()` if one of the above options are changed.

This is needed because the newly introduced `shared_with_groups` is updated on its own below.

Reply:
Thanks @pdecat, it's exactly the reason why this code was needed.